packages/abstract-core/src/objects/fee.rs

```diff
@@ -71,6 +71,76 @@ impl Fee {
     }
 }
 
+/// A wrapper around Coin to help handle fixed fees (with multiples).
+#[cosmwasm_schema::cw_serde]
+pub struct FixedFee {
+    /// Fee to be paid for a unit of a service
+    fee: Coin,
+}
+
+impl FixedFee {
+    /// Creates a wrapped coin to allow charging a fee
+    pub fn new(fee: &Coin) -> Self {
+        FixedFee { fee: fee.clone() }
+    }
+    /// Allows to collect the fee multiple times
+    /// E.g., for namespaces, you want to charge the number of claimed namespaces times the fee for 1 namespace
+    pub fn quantity(mut self, qty: u128) -> Self {
+        self.fee.amount *= Uint128::from(qty);
+        self
+    }
+
+    /// Validates that the sent funds correspond exactly to the fixed fee
+    pub fn charge(self, msg_info: &MessageInfo) -> AbstractResult<Coin> {
```

Comment:
I feel like this doesn't really charge, it's more of an assertion than anything. 
`assert_payment`? Doc comment should also mention something about the returned data.

`charge_mut` can then just be `charge` as charging something implies deducting it from something.